v3/src/components/data-display/hooks/use-connecting-lines.ts

```diff
@@ -0,0 +1,190 @@
+import { curveLinear, line, select } from "d3"
+import { tip as d3tip } from "d3-v6-tip"
+import { useCallback } from "react"
+import { useDataConfigurationContext } from "./use-data-configuration-context"
+import { PixiBackgroundPassThroughEvent, PixiPoints } from "../../graph/utilities/pixi-points"
+import { t } from "../../../utilities/translation/translate"
+import { isGraphDataConfigurationModel } from "../../graph/models/graph-data-configuration-model"
+import { IConnectingLineDescription, transitionDuration } from "../data-display-types"
+import { useMapModelContext } from "../../map/hooks/use-map-model-context"
+
+interface IMouseOverProps {
+  caseIDs: string[]
+  event: MouseEvent
+  parentAttrName?: string
+  primaryAttrValue?: string
+}
+
+interface IDrawLines {
+  allLineCaseIds: Record<string, string[]>
+  lineGroups: Record<string, IConnectingLineDescription[]>
+  parentAttrID?: string
+  parentAttrName?: string
+  pointColorAtIndex: (index: number) => string
+  showConnectingLines: boolean
+}
+
+interface IPrepareLineProps {
+  cellKey?: Record<string, string>
+  connectingLines: IConnectingLineDescription[]
+  parentAttrID?: string
+  parentAttrName?: string
+  pointColorAtIndex: (index: number) => string
+  showConnectingLines: boolean
+}
+
+interface IProps {
+  connectingLinesActivatedRef: React.MutableRefObject<boolean>
+  connectingLinesSvg: SVGGElement | null
+  pixiPoints?: PixiPoints
+}
+
+export const useConnectingLines = (props: IProps) => {
+  const { connectingLinesSvg, connectingLinesActivatedRef, pixiPoints } = props
+  const dataConfig = useDataConfigurationContext()
+  const dataset = dataConfig?.dataset
+  const connectingLinesArea = select(connectingLinesSvg)
+  const mapModel = useMapModelContext()
+  const clientType = isGraphDataConfigurationModel(dataConfig) ? "graph" : "map"
+
+  const dataTip = d3tip().attr("class", `${clientType}-d3-tip`)
+    .attr("data-testid", `${clientType}-connecting-lines-data-tip`)
+    .html((d: string) => {
+      return `<p>${d}</p>`
+    })
+
+  const handleConnectingLinesClick = useCallback((event: MouseEvent, caseIDs: string[]) => {
+    // In the case of the map, temporarily ignore leaflet clicks to prevent the map click handler
```

Comment:
In both map and graph, shift click on connecting line should add the cases belong to the line to the current selection. I notice that the Shift key is not called out here.